client/lib/media/utils.js

```diff
@@ -535,6 +535,14 @@ const MediaUtils = {
 				extension: MediaUtils.getFileExtension( file ),
 				mime_type: MediaUtils.getMimeType( file )
 			} );
+		} else if ( file.thumbnails ) {
```

Comment:
Would you mind explaining what's going on here? I've reviewed the code but it's not clear to me how whether the file we are trying to upload has thumbnails or not is an indication of it being uploaded from an external source. Is this something related to a specific service?